src/overlays/memory_volume_overlay.ts

```diff
@@ -141,34 +165,43 @@ export class MemoryVolumeOverlay extends GenericSdfgOverlay {
         }
     }
 
-    recursively_shade_sdfg(graph, ctx, ppp, visible_rect) {
+    public recursively_shade_sdfg(
+        graph: DagreSDFG, ctx: CanvasRenderingContext2D, ppp: number,
+        visible_rect: SimpleRect
+    ): void {
         graph.nodes().forEach(v => {
-            const state = graph.node(v);
+            const state: State = graph.node(v);
 
             // If we're zoomed out enough that the contents aren't visible, we
             // skip the state.
-            if (ctx.lod && (ppp >= STATE_LOD || state.width / ppp < STATE_LOD))
+            if ((ctx as any).lod && (
+                ppp >= SDFV.STATE_LOD || state.width / ppp < SDFV.STATE_LOD
+            ))
                 return;
 
             // If the node's invisible, we skip it.
-            if (ctx.lod && !state.intersect(visible_rect.x, visible_rect.y,
-                visible_rect.w, visible_rect.h))
+            if ((ctx as any).lod && !state.intersect(
+                visible_rect.x, visible_rect.y,
+                visible_rect.w, visible_rect.h
+            ))
                 return;
 
             const state_graph = state.data.graph;
             if (state_graph && !state.data.state.attributes.is_collapsed) {
-                state_graph.nodes().forEach(v => {
-                    const node = state_graph.node(v);
+                state_graph.nodes().forEach((v: any) => {
+                    const node: SDFGNode = state_graph.node(v);
 
                     // Skip the node if it's not visible.
-                    if (ctx.lod && !node.intersect(visible_rect.x,
-                        visible_rect.y, visible_rect.w, visible_rect.h))
+                    if ((ctx as any).lod && !node.intersect(
```

Comment:
why do we need `ctx as any`?

Reply:
`CanvasRenderingContext2D` doesn't have an attribute `lod`. It's one we force onto it, so TS doesn't know of its existence.